plugins/modules/system/puppet.py

```diff
@@ -87,6 +88,11 @@
     description:
       - Enable full debugging.
     type: bool
+  show_diff:
+    type: bool
+    description:
+      - Internal code, do not use
```

Comment:
I traced this option back to https://github.com/ansible/ansible/commit/badef2912b7a32197de583cfc41f8875516df485. It looks like it was a predecessor of the global `--diff` feature, but never properly converted. I don't think this should be documented, but probably this option should be deprecated and replaced by proper diff support instead.

In any case, the documentation of it should **not** be added in this PR, but we have to keep the corresponding ignore.txt line instead.

This is something to be resolved in another PR :)